### PR TITLE
Build edxapp image and push to docker hub from Travis

### PR DESCRIPTION
**Pull request by thraxil**
This should push the `edxapp` docker image to `appsembler/edxapp:hawthorn` after a successful merge to `appsembler/hawthorn/master`.

The docker hub credentials are in TravisCI's settings for the repo. I tested this out by doing the build/push step in the main `script` section and it seems to work fine.

The only real thing to be aware of with this is that it takes a good 25 minutes to build and push the image. So people will need to understand that they can't `docker pull` immediately after a PR is merged and expect to get the new image. But this should still be better than requiring one of us to do it manually every time.

Some questions for discussion:

* is the `hawthorn` tag the only one we care about? should we also make sure we push to `latest`? (devstack uses `hawthorn`, right?)
* we could pretty easily have it also push an image for each PR tagged with the branch name or SHA1. I left that out for now but is that something anyone else wants? It just adds a long delay to the time Travis takes to run the tests and report success back to the PR.

### CODE REVIEW

**Review by OmarIthawi — COMMENTED** [review]
~That's great! Let me know when this is ready for review. I've added a comment:~

File: .travis.yml [hunk 1]
```diff
@@ -35,3 +36,12 @@ script:
   - docker --version
   - make --version
   - travis_wait 50 make --keep-going $MAKE_TARGET SHARD=$SHARD SHARDS=$SHARDS
+
+# Appsembler addition to automatically push
+# our hawthorn image
+after_success:
+  - if [[ "$TRAVIS_PULL_REQUEST" == "false" && "$TRAVIS_BRANCH" == "appsembler/hawthorn/master" ]]; then
+      docker login -u $DOCKERUSER -p $DOCKERPASSWORD ;
+      travis_wait 50 docker build -f docker/build/edxapp/Dockerfile . -t appsembler/edxapp:hawthorn ;
```
Comment (OmarIthawi) [inline .travis.yml]:
```suggestion
      travis_wait 50 docker build -f docker/build/edxapp/Dockerfile . -t appsembler/edxapp:latest ;
```

File: .travis.yml [hunk 2]
```diff
@@ -35,3 +36,12 @@ script:
...
+      docker push appsembler/edxapp:hawthorn ;
```
Comment (OmarIthawi) [inline .travis.yml]:
```suggestion
      docker push appsembler/edxapp:latest ;
```

**Review by OmarIthawi — COMMENTED** [review]
That's great! Let me know when this is ready for review. I think we should use `latest` for now since that's what my docker devstack has:

```
$ docker images | grep apps
appsembler/edxapp      latest              6e0ad5611c02        2 months ago        2.76GB
```

Not sure how do we get `hawthorn`.

**Comment by thraxil** [comment]
OK. Switched it to `latest`. The only problem with that is that we will have conflicts once we are also dealing Ironwood, etc.

**Comment by OmarIthawi** [comment]
@thraxil either `lastest` or `hawthorn.master` works fine, but let's worry now about Hawthorn. Fixing that minor issue for Ironwood is a tax that I'd be happy to pay when that day comes -- the day when we want to migrate form Hawthorn to Ironwood.

**Pull request merged**